ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/KsqlTestingTool.java

```diff
@@ -35,81 +43,95 @@ private KsqlTestingTool() {
 
   private static final ObjectMapper OBJECT_MAPPER = JsonMapper.INSTANCE.mapper;
 
-  private static int totalNumberOfTests;
-
   public static void main(final String[] args) throws IOException {
-    loadAndRunTests(args);
-  }
 
-  static void loadAndRunTests(final String[] args) {
-    totalNumberOfTests = 0;
-    final List<String> passedTests = new ArrayList<>();
-    final List<Pair<String, String>> failedTests = new ArrayList<>();
     try {
-
       final TestOptions testOptions = TestOptions.parse(args);
-
       if (testOptions == null) {
         return;
       }
-
-      final QttTestFile qttTestFile = OBJECT_MAPPER.readValue(
-          new File(testOptions.getTestFile()), QttTestFile.class);
-      for (final TestCaseNode testCaseNode: qttTestFile.tests) {
-        final List<TestCase> testCases = testCaseNode.buildTests(
-            new File(testOptions.getTestFile()).toPath(),
-            TestFunctionRegistry.INSTANCE.get());
-        for (final TestCase testCase: testCases) {
-          executeTestCase(
-              testCase,
-              new TestExecutor(),
-              passedTests,
-              failedTests);
-        }
+      if (testOptions.getStatementsFile() != null
+          && testOptions.getInputFile() != null
+          && testOptions.getOutputFile() != null) {
+        runWithThripleFiles(
+            testOptions.getStatementsFile(),
+            testOptions.getInputFile(),
+            testOptions.getOutputFile());
       }
-      printResults(totalNumberOfTests, passedTests, failedTests);
-    } catch (final Exception e) {
-      System.err.println("Failed to start KSQL testing tool: " + e.getMessage());
+    } catch (final IOException e) {
+      System.err.println("Invalid arguments: " + e.getMessage());
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private static List<RecordNode> getRecordNodesFromFile(final String filePath) throws IOException {
+    return OBJECT_MAPPER.readValue(new File(filePath), List.class);
+  }
+
+  private static List<String> getSqlStatements(final String queryFilePath) {
+    try {
+      final String sqlStatements = new String(java.nio.file.Files.readAllBytes(
+          Paths.get(queryFilePath)), StandardCharsets.UTF_8);
+
+      final KsqlParser ksqlParser = new DefaultKsqlParser();
+      final List<ParsedStatement> parsedStatements = ksqlParser.parse(sqlStatements);
```

Comment:
just wondering, why do we worry about parsing it here instead of just passing it down and letting it go through the normal path (maybe just split by `;`)

Reply:
Didn't want to change `TestCase` class since it's shared with the QTT.
A simple split by `;` wouldn't work since you can have `;` in constant string fields in expressions which would result in incorrect splitting.